content/id/docs/concepts/extend-kubernetes/api-extension/custom-resources.md

```diff
@@ -0,0 +1,221 @@
+| Kamu bersedia menerima pembatasan format yang Kubernetes terapkan pada jalur *resource* API (Lihat [Ikhtisar API](/docs/concepts/overview/kubernetes-api/).) | Kamu perlu memiliki jalur REST spesifik agar menjadi cocok dengan REST API yang telah didefinisikan. |
+| Sumber data kamu secara alami mencakup hingga sebuah klaster atau sebuah *namespace* dari sebuah klaster. | Sumber daya yang mencakup klaster atau *namespace* adalah sebuah ketidakcocokan; kamu perlu mengendalikan jalur *resource* spesifik. |
+| Kamu ingin menggunakan kembali [dukungan fitur Kubernetes API](#common-features).  | Kamu tidak membutuhkan fitur tersebut. |
+
+### API Deklaratif
+
+Dalam sebuah API Deklaratif, biasanya:
+
+- API kamu terdiri dari sejumlah kecil dari objek yang berukuran relatif kecil (*resource*).
+- Objek-objek mendefinisikan pengaturan dari aplikasi atau infrastruktur.
+- Objek-objek relatif tidak sering diperbaharui.
+- Manusia sering diperlukan untuk membaca dan menulis objek-objek tersebut.
+- Operasi utama terhadap objek bersifat CRUD (*creating, reading, updating,* dan *deleting*).
+- Transaksi antar objek tidak dibutuhkan; API merepresentasikan sebuah keadaan yang diinginkan, bukan keadaan yang eksak.
+
+API imperatif tidak deklaratif.
+Tanda-tanda apabila API kamu tidak deklaratif termasuk:
+- klien berkata "lakukan ini", dan kemudian mendapat sebuah respon serempak ketika selesai.
+- klien berkata "lakukan ini", dan kemudian mendapat sebuah ID operasi kembali, dan harus melakukan sebuah cek terhadap objek Operation terpisah untuk menentukan selesainya sebuah permintaan.
+- Kamu berbicara tentang Remote Procedure Call (RPC).
+- Menyimpan secara langsung sejumlah data (mis. > beberapa kB per objek, atau >1000-an objek).
+- Membutuhkan akses dengan bandwidth tinggi (10-an permintaan per detik dapat ditopang).
+- Menyimpan data pengguna (seperti gambar, PII, dll) atau data berskala besar yang diproses oleh aplikasi.
+- Operasi-operasi natural terhadap objek yang tidak bersifat CRUD.
+- API yang tidak mudah dimodelkan dengan objek.
+- Kamu memilih untuk merepresentasikan operasi tertunda dengan sebuah ID operasi atau sebuah objek operasi.
+
+## Apakah saya harus menggunakan sebuah configMap atau sebuah Custom Resource?
+
+Gunakan ConfigMap jika salah satu hal berikut berlaku:
+* Terdapat sebuah format berkas pengaturan yang sudah ada, terdokumentasi dengan baik seperti sebuah `mysql.cnf` atau `pom.xml`.
+* Kamu ingin menaruh seluruh berkas pengaturan kedalam sebuah *key* dari sebuah configMap.
+* Kegunaan utama dari berkas pengaturan adalah untuk dikonsumsi sebuah program yang berjalan di dalam sebuah Pod di dalam klaster kamu untuk mengatur dirinya sendiri.
+* Konsumen dari berkas lebih suka untuk mengkonsumsi lewat berkas dalam sebuah Pod atau variabel lingkungan dalam sebuah Pod, dibandingkan melalui Kubernetes API.
+* Kamu ingin melakukan pembaharuan bergulir lewat Deployment, dll, ketika berkas diperbaharui.
```

Comment:
this is a really nice translation! we should translate all "rolling update" to "pembaharuan bergulir" in the translated pages as well

wdyt @irvifa @phanama @danninov @wahyuoi ?